serialization/compact_config.go

```diff
@@ -51,3 +63,55 @@ func (cc *CompactConfig) ensureSerializers() {
 		cc.serializers = make(map[string]CompactSerializer)
 	}
 }
+
+func (cc *CompactConfig) checkNoDefaultSerializer() error {
+	// note that the check we are doing is just information.
+	// the default serializers are looked uo before the compact serializer,
+	// so the user cannot override any default serializer.
```

Comment:
this part is different from the java implementation and the order defined in https://docs.hazelcast.com/hazelcast/latest/serialization/serialization#steps. why this is the case in go client?